### PR TITLE
Modify annotations to support model predictions

### PR DESCRIPTION
**Pull request by gregorjerse**

### CODE REVIEW

File: resolwe/flow/models/annotations.py [hunk 1]
```diff
@@ -295,8 +289,12 @@ class AnnotationField(models.Model):
     #: field version
     version = VersionField(number_bits=VERSION_NUMBER_BITS, default="0.0.0")
 
+    #: the latest version of the fields
```
Comment (dblenkus) [inline resolwe/flow/models/annotations.py:292]: These are objects, not fields.

File: resolwe/flow/models/annotations.py [hunk 2]
```diff
@@ -295,8 +289,12 @@ class AnnotationField(models.Model):
...
     objects = AnnotationFieldManager()
 
+    #: all fields
```
Comment (dblenkus) [inline resolwe/flow/models/annotations.py:295]: Same here.

File: resolwe/flow/models/entity.py [hunk 3]
```diff
@@ -367,41 +371,47 @@ def update_annotations(self, annotations: dict[str, Any], contributor, update=Tr
...
+            annotation_values.append(value)
+            try:
+                value.validate()
+            except ValidationError as e:
+                validation_errors.append(e)
         if validation_errors:
             raise ValidationError(validation_errors)
 
-        # Delete and update annotations in a transaction.
+        # Whin this is bulk set also delete other annotation values by setting delete
```
Comment (dblenkus) [inline resolwe/flow/models/entity.py:393]: whin -> when

File: resolwe/flow/tests/test_ordering.py [hunk 4]
```diff
@@ -28,13 +28,20 @@ def setUp(self):
...
     def test_ordering_version(self):
-        response = self.client.get(self.url, {"ordering": "version"}, format="json")
-        self.assertEqual(response.data[0]["id"], self.proc_1.id)
-        self.assertEqual(response.data[1]["id"], self.proc_2.id)
-
-        response = self.client.get(self.url, {"ordering": "-version"}, format="json")
-        self.assertEqual(response.data[0]["id"], self.proc_2.id)
-        self.assertEqual(response.data[1]["id"], self.proc_1.id)
+        pass
+        # TODO: this test makes no sense. Only the latest version is returned by the
```
Comment (dblenkus) [inline resolwe/flow/tests/test_ordering.py:32]: Then remove it :)

File: resolwe/flow/models/annotations.py [hunk 5]
```diff
@@ -454,6 +454,15 @@ class Meta:
         ]
         ordering = ["field__group__sort_order", "field__sort_order"]
 
+    #: URL slug
+    slug = ResolweSlugField(
+        populate_from=lambda value: (
+            f"{value.entity.slug}-{value.field.group.name}" f"-{value.field.name}"
```
Comment (dblenkus) [inline resolwe/flow/models/annotations.py]: Can you join this into a single sting?

File: resolwe/flow/models/annotations.py (hunk 1 above)
Comment (gregorjerse) [reply resolwe/flow/models/annotations.py:295]: Done.

File: resolwe/flow/models/annotations.py (hunk 2 above)
Comment (gregorjerse) [reply resolwe/flow/models/annotations.py:298]: Done.

File: resolwe/flow/models/entity.py (hunk 3 above)
Comment (gregorjerse) [reply resolwe/flow/models/entity.py:393]: Done.

File: resolwe/flow/models/annotations.py (hunk 5 above)
Comment (gregorjerse) [reply resolwe/flow/models/annotations.py]: This is different now due to migration problems, they do not like lambdas.

**Review by dblenkus — APPROVED** [review]

**Pull request merged**